### PR TITLE
Update Checkout feature in DeveloperGuide.md

### LINKED ISSUE #180: Update Checkout feature in Developer Guide for v1.3
Checkout feature in Developer Guide (DG) is out of date. 
We need to update the Checkout feature in the DG for v1.3.

Things we need to update:
1. Feature description.
2. State Diagrams.
3. Sequence Diagrams.
4. Actvity Diagrams.

### PR DESCRIPTION
**Pull request by deeyonn**
Resolves #180 

* Updated feature description.
* Modified state diagrams.
* Added sequence diagrams.
* Added activity diagrams.
* Fix several puml issues.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -238,62 +238,64 @@ _{more aspects and alternatives to be added}_
 
 _{Explain here how the data archiving feature will be implemented}_
 
-### \[Proposed\] Branch feature
+### Check-out feature
```
Comment (cliftonfelix) [inline docs/DeveloperGuide.md:241]: Since the command name is checkout, is it better to remove the hyphen here?

**Review by lulucopter — APPROVED** [review]
LGTM.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (deeyonn) [reply docs/DeveloperGuide.md:241]: Changed spelling of Check-out to checkout

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -238,62 +238,64 @@ _{more aspects and alternatives to be added}_
...
 
-### \[Proposed\] Branch feature
+### checkout feature
 
-#### Proposed Implementation
+#### Implementation
 
-The proposed branching mechanism is facilitated by `BranchStorage`. It extends `Storage` with a branch history,
-stored internally as an `branchList` and `currentBranchPointer`. Additionally, it implements the following operations:
+The check-out mechanism is facilitated by `CheckoutCommand` that extends `Command`. It is supported by `CheckoutCommandParser` and `AddressBookFile`.
```
Comment (cliftonfelix) [inline docs/DeveloperGuide.md:245]: Here also

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -238,62 +238,64 @@ _{more aspects and alternatives to be added}_
...
 
-These operations are exposed in the `Model` interface as `Model#createBranch()`, `Model#checkoutBranch()` and
-`Model#resetBranch()` respectively.
+* `CheckoutCommand#execute()` — Executes the command to switch to a particular list stored in the `data` folder parsed from the `userInput` using `CheckoutCommandParser#parse()`.
+* `CheckoutCommand#equals()` — Checks whether an instance of a `CheckoutCommand` is equal to another, by checking:
+    - Whether they are the same instance
+    - Whether the specified list is the same in two different instances
 
-Given below is an example usage scenario and how the branching mechanism behaves at each step.
+Given below is an example usage scenario and how the check-out mechanism behaves at each step.
```
Comment (cliftonfelix) [inline docs/DeveloperGuide.md:254]: And here

**Review by cliftonfelix — APPROVED** [review]
LGTM

**Pull request merged**